common/engine-entities.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "components.pb.h"
+#include "event-system.hpp"
+#include "events.hpp"
+
+namespace tec {
+
+enum ENGINE_ENTITIES { MANIPULATOR = 1 };
+
+namespace { // Make the follow functions header local so they can't be invoked directly.
+void CreateManipulatorEntity() {
+	proto::Entity entity;
+	entity.set_id(ENGINE_ENTITIES::MANIPULATOR);
+	entity.add_components()->mutable_renderable();
+	EventSystem<EntityCreated>::Get()->Emit(std::make_shared<EntityCreated>(EntityCreated{entity}));
+}
+} // namespace
+
+inline void CreateEngineEntities() { CreateManipulatorEntity(); }
```

Comment:
I don't exactly like stand alone functions in headers, it's very messy.
The functions could just as easily go in client/game.cpp, which doesn't require an inline and a pocket dimension.

Reply:
My goal was to remove this logic from game.cpp to reduce file size. Ideally these would be in a list of entities that area loaded in.

Reply:
i would say loading from game.cpp aswell. Having it in one place would make it less troublesome to debug in such a situation

Reply:
Moved.